security/pkg/k8s/controller/monitoring.go

```diff
@@ -15,78 +15,78 @@
 package controller
 
 import (
-	"github.com/prometheus/client_golang/prometheus"
+	"istio.io/istio/security/pkg/monitoring"
 )
 
 const (
 	errorlabel = "error"
 )
 
 var (
-	serviceAccountCreationCounts = prometheus.NewCounterVec(prometheus.CounterOpts{
-		Namespace: "citadel",
-		Subsystem: "secret_controller",
-		Name:      "svc_acc_created_cert_count",
-		Help:      "The number of certificates created due to service account creation.",
-	}, []string{})
+	errorTag     = monitoring.MustCreateTag(errorlabel)
+	subsystemTag = monitoring.MustCreateTag("subsystem")
```

Comment:
In Prometheus, Namespace, Subsystem, etc. are just a way to fully name the metric by joining them using _

Reply:
Also, why do we need to create a tag for this? This metric is Citadel scoped, so I think it's okay to not have a tag for it?

Reply:
As I understand it, the value passed in here isn't the value for the tag, but rather the name of the key. The value should indeed be the `secret_controller`, and [that is set here](https://github.com/istio/istio/pull/15223/files#diff-20057f9306a9c65d483295b9aa6d0ea3R80). I'm not entirely sure if the `subsystem` label is needed-- it is Citadel scoped, but the label is to differentiate it from metrics in other Citadel components (like the `sds_service` metrics added [here](https://github.com/istio/istio/pull/14902/files#diff-f830b3497f81e72ed4149fa33b393442R33)).

Reply:
1.  I don't think we will need tag for `secret_controller` here, unless in Citadel we have the same metric for more than one dimension/tag. In either case, I don't think we need to include `subsystemTag` every time you create a new metric in the global variable creation, since `With` is also creating a new metric.

2. Node agent is running as a different deployment so I don't think it will have any conflicts.

But you can test it by port-forwarding to 15014 and see the metrics.

Reply:
Ah I see, I think you're right on this. Removed the `subsystem` tag :)